src/auth/auth.service.ts

```diff
@@ -59,6 +61,34 @@ export class AuthService {
     };
   }
 
+  async loginSteam(loginDto: CreateSteamUserDto) {
+    let user = await this.steamUserService.findOneBySteamId(loginDto.steamid);
+
+    if (!user) {
+      user = await this.steamUserService.create(loginDto);
+    }
+
+    const tokens = await this.issueTokens(user.id);
+
+    return {
+      user,
+      ...tokens,
+      jwtExpiresIn: getTime(addSeconds(new Date(), JWT_EXPIRES_IN_SECONDS)),
+    };
+  }
+
+  async findOne(id: string) {
+    const steamUser = await this.steamUserService.findOne(id);
+
+    if (!steamUser) {
+      const user = await this.userService.findUnique({ id });
+
+      return user;
+    }
+
+    return steamUser;
+  }
```

Comment:
This function should look only in `userService`.
```suggestion
  async findOne(id: string) {
    return this.userService.findUnique({ id });
  }
```